php_phongo.c

```diff
@@ -1963,6 +1975,14 @@ static mongoc_ssl_opt_t* php_phongo_make_ssl_opt(zval* zoptions TSRMLS_DC)
 
 	ssl_opt = ecalloc(1, sizeof(mongoc_ssl_opt_t));
 
+	/* If SSL options are set in the URL, we need to read them and set them on
+	 * the options struct so we can merge potential options from passed in
+	 * driverOptions (zoptions)*/
+	if (mongoc_uri_get_ssl(uri)) {
```

Comment:
In https://github.com/mongodb/mongo-php-driver/pull/893/files#r204696800, you said this may no longer be necessary; however, I think without it you'd always end up assigning `any_ssl_option_set` to true.

As-is, there's still an edge case where `any_ssl_option_set` is true and we still end up returning an unmodified `mongoc_ssl_opt_t` struct. `mongoc_uri_get_ssl()` may return true even if only the `ssl=true` URI option is set. In that case, `php_phongo_mongoc_ssl_opts_from_uri()` isn't really copying anything and simply assigning defaults and we return an empty struct. Assigning the empty struct is a pointless copy and will redundantly enable SSL on the client.

I suppose a more correct approach would be passing `&any_ssl_option_set` into `php_phongo_mongoc_ssl_opts_from_uri()` so that it can be set only if actual values are copied (i.e. a non-null `const char *` or a true value for one of the two boolean options). _Then_, you could remove `mongoc_uri_get_ssl()` and guarantee that this function returns `NULL` if the struct is truly never modified (i.e. non-zero).

Reply:
I meant for: https://github.com/mongodb/mongo-php-driver/pull/893/files#diff-c06c6e1c9374aecabcf544157f9d0c26R2479

Reply:
Assuming `php_phongo_mongoc_ssl_opts_from_uri()` will need to be updated for https://github.com/mongodb/mongo-php-driver/pull/893#discussion_r204735780, can we also pass in `&any_ssl_option_set` so it's only set when a struct field is actually assigned? That would still remove this conditional.

Alternatively, I suppose the code from `php_phongo_mongoc_ssl_opts_from_uri()` can simply be inlined in this function.